integration-tests/src/test/java/org/apache/fineract/integrationtests/ClientExternalIdTest.java

```diff
@@ -0,0 +1,320 @@
+        // then
+        assertNotNull(clientResponse);
+        assertNotNull(clientResponse.getResourceExternalId());
+        assertEquals(externalId, clientResponse.getResourceExternalId());
+
+        fetchClientByExternalId(clientResponse.getResourceExternalId());
+
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, false);
+    }
+
+    @Test
+    public void testClientStatusUsingExternalId() {
+        ClientHelper clientHelper = new ClientHelper(requestSpec, responseSpec);
+        GlobalConfigurationHelper.manageConfigurations(requestSpec, responseSpec,
+                GlobalConfigurationHelper.ENABLE_AUTOGENERATED_EXTERNAL_ID, true);
+        String jsonPayload = ClientHelper.getBasicClientAsJSON(ClientHelper.DEFAULT_OFFICE_ID, ClientHelper.LEGALFORM_ID_PERSON, null);
+        final PostClientsResponse addClientResponse = ClientHelper.addClientAsPerson(requestSpec, responseSpec, jsonPayload);
+        final String clientExternalId = addClientResponse.getResourceExternalId();
+        final Long clientId = addClientResponse.getClientId();
+        assertNotNull(clientExternalId);
+        log.info("Client data id {} and external Id {}", clientId, clientExternalId);
+
+        GetClientsClientIdResponse clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.ACTIVE, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Close Client action
+        jsonPayload = clientHelper.getCloseClientAsJSON();
+        PostClientsClientIdResponse commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec,
+                clientExternalId, ClientHelper.CLOSE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        assertNotNull(commandResponse.getResourceExternalId());
+        log.info("Client data id {} and external Id {}", commandResponse.getResourceId(), clientExternalId);
+        assertEquals(clientId.intValue(), commandResponse.getResourceId());
+
+        clientResponse = ClientHelper.getClientByExternalId(requestSpec, responseSpec, clientExternalId);
+        ClientStatusChecker.verifyClientStatus(ClientStatus.CLOSED, clientResponse);
+        log.info("Client data id {} and status {}", clientExternalId, clientResponse.getStatus().getCode());
+
+        // Reactivate Client action
+        jsonPayload = clientHelper.getReactivateClientAsJSON();
+        commandResponse = ClientHelper.performClientActionUsingExternalId(requestSpec, responseSpec, clientExternalId,
+                ClientHelper.REACTIVATE_CLIENT_COMMAND, jsonPayload);
+        assertNotNull(commandResponse);
+        assertNotNull(commandResponse.getResourceExternalId());
```

Comment:
You know the value of the external id, so please do equality check instead of null check here

Reply:
Done